cmd/upgrade/roles/cmd.go

```diff
@@ -0,0 +1,1012 @@
+	if reporter.IsTerminal() {
+		spin = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+	}
+	if spin != nil {
+		spin.Start()
+	}
+	if !args.isInvokedFromClusterUpgrade {
+		reporter.Infof("Ensuring account role/policies compatibility for upgrade")
+	}
+
+	//ACCOUNT ROLES
+
+	isUpgradeNeedForAccountRolePolicies, err := awsClient.IsUpgradedNeededForAccountRolePoliciesForCluster(
+		cluster,
+		policyVersion,
+	)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		LogError(roles.RosaUpgradeAccRolesModeAuto, ocmClient, policyVersion, err, reporter)
+		os.Exit(1)
+	}
+
+	if spin != nil {
+		spin.Stop()
+	}
+
+	if !isUpgradeNeedForAccountRolePolicies {
+		reporter.Infof("Account roles/policies for cluster '%s' are already up-to-date.", r.ClusterKey)
+	} else {
+		accountRolePolicies, err := ocmClient.GetPolicies("")
+		if err != nil {
+			reporter.Errorf("Expected a valid role creation mode: %s", err)
+			os.Exit(1)
+		}
+
+		switch mode {
+		case aws.ModeAuto:
+			if isUpgradeNeedForAccountRolePolicies {
+				reporter.Infof("Starting to upgrade the policies")
+				err = upgradeAccountRolePoliciesFromCluster(
+					mode,
+					reporter,
+					awsClient,
+					cluster,
+					creator.AccountID,
+					accountRolePolicies,
+					policyVersion,
+				)
+				if err != nil {
+					LogError(roles.RosaUpgradeAccRolesModeAuto, ocmClient, policyVersion, err, reporter)
+					if args.isInvokedFromClusterUpgrade {
+						return err
+					}
+					reporter.Errorf("Error upgrading the role polices: %s", err)
+					os.Exit(1)
+				}
+			}
+		case aws.ModeManual:
+			err = aws.GeneratePolicyFiles(reporter, env, isUpgradeNeedForAccountRolePolicies,
+				false, accountRolePolicies, nil)
+			if err != nil {
+				reporter.Errorf("There was an error generating the policy files: %s", err)
+				os.Exit(1)
+			}
+			if reporter.IsTerminal() {
+				reporter.Infof("All policy files saved to the current directory")
+				reporter.Infof("Run the following commands to upgrade the account role policies:\n")
+			}
+
+			commands, err := buildAccountRoleCommandsFromCluster(
+				mode,
+				cluster,
+				creator.AccountID,
+				isUpgradeNeedForAccountRolePolicies,
+				awsClient,
+				policyVersion,
+			)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(commands)
+			if args.isInvokedFromClusterUpgrade {
+				reporter.Infof("Run the following command to continue scheduling cluster upgrade"+
+					" once account and operator roles have been upgraded : \n\n"+
+					"\trosa upgrade cluster --cluster %s\n", r.ClusterKey)
+				return nil
+			}
+
+		default:
+			reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
+			os.Exit(1)
+		}
+	}
+
+	//OPERATOR ROLES
+	if !isInvokedFromClusterUpgrade {
+		reporter.Infof("Ensuring operator role policies compatibility for upgrade")
+	}
+
+	if spin != nil {
+		spin.Start()
+	}
+
+	operatorRoles, hasOperatorRoles := cluster.AWS().STS().GetOperatorIAMRoles()
+	if !hasOperatorRoles || len(operatorRoles) == 0 {
```

Comment:
Should we exit the program in this case?
If I'm not wrong for `len(operatorRoles) == 0` we will get this error from `GetOperatorPaths`:
`"Can not detect operator policy path. Existing operator roles do not have operator policies attached to them"`

Reply:
Original upgrade operator also didn't, but I guess it makes sense